src/Services/Azure/Monitor/MonitorService.cs

```diff
@@ -197,21 +196,44 @@ public async Task<List<JsonNode>> QueryLogs(
 
         try
         {
-            // Convert hours to days for QueryWorkspace
-            double days = (hours ?? 24) / 24.0;
+            var credential = await GetCredential(tenant);
+            var options = AddDefaultPolicies(new LogsQueryClientOptions());
+
+            if (retryPolicy != null)
+            {
+                options.Retry.Delay = TimeSpan.FromSeconds(retryPolicy.DelaySeconds);
+                options.Retry.MaxDelay = TimeSpan.FromSeconds(retryPolicy.MaxDelaySeconds);
+                options.Retry.MaxRetries = retryPolicy.MaxRetries;
+                options.Retry.Mode = retryPolicy.Mode;
+                options.Retry.NetworkTimeout = TimeSpan.FromSeconds(retryPolicy.NetworkTimeoutSeconds);
+            }
```

Comment:
If retry policy is set, are all these properties required to be set as well? If not, should we continue to keep the defaults and update only non-null values? For e.g. the user may just want to update the `MaxDelay` and not set the `Delay` in retry policy options.

Reply:
What do you recommend?

Reply:
Since these are all value types and it's just setting to the default, I think any check like that would be a breaking change to make them nullable or not be worth the effort.

Reply:
Makes sense, thanks for checking.